### PR TITLE
Add initial ideas from openCDE Hackathon

### PR DESCRIPTION
**Pull request by GeorgDangl**
This is an early draft for inclusion of document metadata descriptions.

### CODE REVIEW

**Comment by GeorgDangl** [comment]
This is not yet ready to be merged😊

File: openapi/openCDE-spec.yaml [hunk 1]
```diff
@@ -313,4 +322,43 @@ components:
...
+          type: boolean
+        enumValues:
+          type: array
+          items:
+            type: string
+        arrayType:
+          $ref: '#/components/schemas/DataType'
+      required: ['type', 'required']
+    
+    DataTypeWithName:
```
Comment (pasi-paasiala) [inline openapi/openCDE-spec.yaml]: I wonder if we should still leave this out from this commit, since it is not referred from anywhere.

Comment (LionelOnTech) [reply openapi/openCDE-spec.yaml]: I agree. Just having the DataType would be good for now. Till we have a use for the other type.

**Comment by GeorgDangl** [comment]
@LionelOnTech, I just checked with the OpenAPI definitions, `date-time` (with a hyphen) should be correct, so I think we can keep it here in the enum.  
See https://swagger.io/docs/specification/data-models/data-types/#format

**Pull request merged**